src/scales/scale.category.js

```diff
@@ -22,15 +22,36 @@ function findOrAddLabel(labels, raw, index, addedLabels) {
 
 const validIndex = (index, max) => index === null ? null : _limitValue(Math.round(index), 0, max);
 
+function getLabelForValue(value) {
+  const labels = this.getLabels();
+
+  if (value >= 0 && value < labels.length) {
+    return labels[value];
+  }
+  return value;
+}
+
 export default class CategoryScale extends Scale {
 
+  static id = 'category';
+
+  /**
+   * @type {any}
+   */
+  static defaults = {
+    ticks: {
+      callback: getLabelForValue
+    }
+  };
+
   constructor(cfg) {
     super(cfg);
 
     /** @type {number} */
     this._startValue = undefined;
     this._valueRange = 0;
     this._addedLabels = [];
+    this.getLabelForValue = getLabelForValue;
```

Comment:
This may have a subtle change in behaviour if someone creates a scale that extends this class and overrides `getLabelForValue`

Reply:
@etimberg fixed